AutoMad/wordGrabber.py

```diff
@@ -0,0 +1,147 @@
+    # created automatically when the authorization flow completes for the first
+    # time.
+    if os.path.exists("token.json"):
+        creds = Credentials.from_authorized_user_file("token.json", SCOPES)
+    # If there are no (valid) credentials available, let the user log in.
+    if not creds or not creds.valid:
+        if creds and creds.expired and creds.refresh_token:
+            creds.refresh(Request())
+        else:
+            flow = InstalledAppFlow.from_client_secrets_file(
+                "credentials.json", SCOPES
+            )
+            creds = flow.run_local_server(port=0)
+        # Save the credentials for the next run
+        with open("token.json", "w") as token:
+            token.write(creds.to_json())
+    return creds
+
+def getStoryByName(name):
+    result = getGoogleSheet(MADLIB_SHEET_ID,STORIES)    
+    values=result['values']
+    df = pd.DataFrame(values)
+    df.columns = df.iloc[0]
+    df = df[1:]
+    for index,row in df.iterrows():
+        if row['storyname'] == name:
+            return row['urlString']
+    return None
+
+def getStories():
+    result = getGoogleSheet(MADLIB_SHEET_ID,STORIES)
+    values=result['values']
+    df = pd.DataFrame(values)
+    df.columns = df.iloc[0]
+    df = df[1:]
+    return df.storyname
+    
+def getStory(docID):
+    creds = getCreds()
+    try:
+        service = build("docs", "v1", credentials=creds)
+        # Retrieve the documents contents from the Docs service.
+        document = service.documents().get(documentId=docID, includeTabsContent=True).execute()
+        plainText=''
+        allTabs = get_all_tabs(document)
+        for tab in allTabs:
+           # Get the DocumentTab from the generic Tab.
+            document_tab = tab.get('documentTab')
+            doc_content = document_tab.get('body').get('content')
+            plainText += read_structural_elements(doc_content)
+        return plainText
+    
+        
+    except HttpError as err:
+        print(err)
+
+#gets the word library from google sheets
+def getGoogleSheet(sheetID, sheetRange): 
+    creds = getCreds()
+    try:
+        service = build("sheets", "v4", credentials=creds)
+        # Call the Sheets API
+        sheet = service.spreadsheets()
+        result = (
+            sheet.values()
+            .get(spreadsheetId=sheetID, range=sheetRange)
+            .execute()
+        )
+        return result
+    except HttpError as err:
+        print(err)
+        return None
+
+def getDictionary():
+    result = getGoogleSheet(MADLIB_SHEET_ID,WORDS)
+    values=result['values']
+    df = pd.DataFrame(values)
+    df.columns = df.iloc[0]
+    df = df[1:]
+    df = df.replace('', np.nan)
+    dictionary = madDictionary.MadDictionary(df)
+    return dictionary
+    
+#code from google documentation to extract the text from a document
+def get_all_tabs(doc):
+  all_tabs = []
+  for tab in doc.get('tabs'):
+    all_tabs.append(tab)
+  return all_tabs
```

Comment:
```suggestion
def get_all_tabs(doc):
  return [tab for tab in doc.get('tabs')]
```

Or you don't really need this function at all, you could just inline the above where you need it.